presto-main-base/src/main/java/com/facebook/presto/server/thrift/ThriftCodecWrapper.java

```diff
@@ -46,13 +54,25 @@ public static <T> ThriftCodec<T> unwrapThriftCodec(Codec<T> codec)
     @Override
     public byte[] toBytes(T instance)
     {
-        throw new UnsupportedOperationException("Operation not supported");
+        try {
+            SliceOutput sliceOutput = new DynamicSliceOutput(1024);
```

Comment:
you can use drift here and avoid airlift. 

example : 
https://github.com/facebookexternal/presto-facebook/blob/master/presto-facebook-hive/src/main/java/com/facebook/presto/hive/HiveMetadataUpdateHandleThriftSerde.java#L47